src/containers/Login.js

```diff
@@ -0,0 +1,95 @@
+import React, { Component, Fragment } from 'react';
+import {
+  Button,
+  Grid,
+  Icon,
+  Image,
+  Message,
+} from 'semantic-ui-react';
+import axios from 'axios';
+import URL from 'url-parse';
+import queryString from 'query-string';
+
+import logoImage from '../assets/images/rovercode_logo.png';
+
+export default class Login extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      error: false,
+    };
+  }
+
+  redirectToSocial = (element) => {
+    const service = element.target.id;
+    return axios.post(`/jwt/auth/social/${service}/auth-server/`)
+      .then((response) => {
+        const url = URL(response.data.url);
+        const parsed = queryString.parse(url.query);
+        const redirectUrl = URL(parsed.redirect_uri);
+        redirectUrl.set('pathname', `/login/${service}/callback/`);
+        parsed.redirect_uri = redirectUrl.toString();
+        url.query = queryString.stringify(parsed);
+        window.location.assign(url.toString());
+      })
+      .catch(() => {
+        this.setState({
+          error: true,
+        });
+      });
+  }
+
+  render() {
+    const { error } = this.state;
+
+    return (
+      <Fragment>
+        <Grid centered columns="equal">
+          <Grid.Row>
+            <Image src={logoImage} />
+          </Grid.Row>
+          <Grid.Row>
+            <p>
+              You can sign up / sign in with one of your existing third-party accounts.
```

Comment:
Will we add the non-third-party login (just a native rovercode account) login later?

Reply:
Yes. I should have put that in the description. I planned on adding that later but it would also involve a sign up page and I didn't want this to get too big to review.

Reply:
Good call. Smaller steps in PRs are great.